modules/javafx.graphics/src/main/java/javafx/scene/ContainsProperties.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package javafx.scene;
+
+import java.util.HashMap;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableMap;
+
+/**
+ * Indicates that the implementor contains Properties.
+ * @since 22
+ */
+public interface ContainsProperties {
```

Comment:
This name fails the "is-a" test "is a contains properties", like for example "is an `EventTarget`" or "is a `Styleable`". Suggestions:

`ApplicationPropertyProvider`
`PropertyProvider`
`ApplicationPropertyAccessor`
`PropertyAccessor`

I'd personally go for the most descriptive one (the first one).  It's long, but it's unlikely to be ever encountered as a type for a variable.

Suggestion for the docs:

    /*
     * Provides access to properties for use primarily by application developers.
     */

Reply:
Shouldn't it be `ApplicationPropertiesProvider` then, since multiple properties are involved?